.github/workflows/build-test.yml

```
@@ -165,7 +165,7 @@ jobs:
           fi
 
       - name: Upload coverage reports to Codecov
-        if: inputs.coverage == 'ON'
+        if: inputs.coverage == 'ON' && github.event.pull_request.head.repo.full_name == github.repository
```

```suggestion
        # PRs from external contributors fail: https://github.com/codecov/feedback/issues/301
        # Only upload coverage reports for PRs from the same repo (not forks)
        if: inputs.coverage == 'ON' && github.event.pull_request.head.repo.full_name == github.repository
```

Just added this in another repo. No sure if you'd like references in comments. Please feel free to ignore.